src/components/AssistantPlainTextModal.vue

```diff
@@ -0,0 +1,258 @@
+<template>
+	<NcModal v-if="show"
+		:size="modalSize"
+		:can-close="false"
+		class="assistant-modal"
+		@close="onCancel">
+		<div ref="modal_content"
+			class="assistant-modal--wrapper">
+			<div class="assistant-modal--content">
+				<span class="assistant-bubble">
+					<CreationIcon :size="16" class="icon" />
+					<span>{{ t('assistant', 'Nextcloud Assistant') }}</span>
+				</span>
+				<h2>
+					{{ taskTypeName }}
+				</h2>
+				<NcButton :aria-label="closeButtonLabel"
+					:title="closeButtonTitle"
+					type="tertiary"
+					class="close-button"
+					@click="onCancel">
+					<template #icon>
+						<CloseIcon />
+					</template>
+				</NcButton>
+				<div v-if="myOutput"
+					class="output">
+					<label
+						for="assistant-output"
+						class="input-label">
+						{{ t('assistant', 'Output') }}
+					</label>
+					<NcRichContenteditable
+						id="assistant-output"
+						ref="output"
+						:value.sync="myOutput"
+						class="editable-output"
+						:multiline="true"
+						:placeholder="t('assistant', 'Output')"
+						:link-autocomplete="false" />
+					<NcNoteCard
+						class="warning-note"
+						type="warning">
+						{{ t('assistant', 'This output was generated by AI. Make sure to double-check and adjust.') }}
+					</NcNoteCard>
+					<div class="button-wrapper">
+						<NcButton type="secondary"
+							:title="t('assistant', 'Copy output text to clipboard')"
+							@click="onCopy">
+							{{ t('assistant', 'Copy output') }}
+							<template #icon>
+								<ClipboardCheckOutlineIcon v-if="copied" />
+								<ContentCopyIcon v-else />
+							</template>
+						</NcButton>
+						<NcButton :disabled="output === myOutput"
+							type="secondary"
+							:title="t('assistant', 'Reset the output value to the originally generated one')"
+							@click="delayedReset">
+							{{ t('assistant', 'Reset') }}
+						</NcButton>
+					</div>
+				</div>
+			</div>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import CloseIcon from 'vue-material-design-icons/Close.vue'
+import CreationIcon from 'vue-material-design-icons/Creation.vue'
+import ClipboardCheckOutlineIcon from 'vue-material-design-icons/ClipboardCheckOutline.vue'
+import ContentCopyIcon from 'vue-material-design-icons/ContentCopy.vue'
+
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcRichContenteditable from '@nextcloud/vue/dist/Components/NcRichContenteditable.js'
+import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
+
+import VueClipboard from 'vue-clipboard2'
+import Vue from 'vue'
+
+import { showError } from '@nextcloud/dialogs'
+
+import { TASK_TYPES } from '../constants.js'
+
+Vue.use(VueClipboard)
+
+export default {
+	name: 'AssistantPlainTextModal',
```

Comment:
Is this component going to be used in other contexts than STT? If not, we could rename it to `SttResultModal`.

Reply:
I must admit, I have no other definite use case for a plain text result page yet but I ventured a guess here that we might have those in the future (for example, notifications where we want to hide which text processing task type was responsible for the result) . Although, it might also be that such features would be otherwise accommodated in the notification logic?

Reply:
Anyway, not a big deal. Rename it if you want, merge if you don't :grin: